test/cuda/test_autotuner_utility.cc

```diff
@@ -46,86 +50,85 @@ TEST(DivisorsAndPowers, Default) {
-  tc::OptionsCache::getCache()->clear();
-}
-
 TEST(RestoreCandidates, NoRuntimeRecorded) {
-  EnableCaches();
+  CudaOptionsCache optionsCache;
   std::vector<at::Tensor> inputs{at::CUDA(at::kFloat).rand({10, 16}),
                                  at::CUDA(at::kFloat).rand({16, 20})};
-
-  tc::ATenCompilationUnit<tc::CudaTcExecutor> atCompl;
-  atCompl.define(tc_);
-  tc::CudaMappingOptions options =
-      tc::CudaMappingOptions::makeMlpMappingOptions();
-  std::vector<at::Tensor> outputs_;
-  auto handle = atCompl.compile("matmul", inputs, options);
-  atCompl.run("matmul", inputs, outputs_, handle);
+  auto options = CudaMappingOptions::makeMlpMappingOptions();
+  auto pExecutor = compile<CudaBackend>(tc_, "matmul", inputs, options);
+  std::vector<at::Tensor> outputs = prepareOutputs(tc_, "matmul", inputs);
+  run(*pExecutor, inputs, outputs);
 
   FLAGS_tuner_gen_restore_number = 1;
-  ASSERT_EQ(restoreCandidates(tc_, inputs, outputs_).size(), 0u);
+  ASSERT_EQ(restoreCandidates(optionsCache, tc_, inputs, outputs).size(), 0u);
 }
 
 TEST(RestoreCandidates, Hit) {
-  EnableCaches();
+  CudaOptionsCache optionsCache;
   std::vector<at::Tensor> inputs{at::CUDA(at::kFloat).rand({10, 16}),
                                  at::CUDA(at::kFloat).rand({16, 20})};
-
-  tc::ATenCompilationUnit<tc::CudaTcExecutor> atCompl;
-  atCompl.define(tc_);
-  tc::CudaMappingOptions options =
-      tc::CudaMappingOptions::makeMlpMappingOptions();
-  std::vector<at::Tensor> outputs_;
-  auto handle = atCompl.compile("matmul", inputs, options);
-  atCompl.run("matmul", inputs, outputs_, handle, true);
-
-  options = tc::CudaMappingOptions::makeNaiveMappingOptions();
-  handle = atCompl.compile("matmul", inputs, options);
-  atCompl.run("matmul", inputs, outputs_, handle, true);
+  auto options = CudaMappingOptions::makeMlpMappingOptions();
+  auto pExecutor = compile<CudaBackend>(tc_, "matmul", inputs, options);
+  std::vector<at::Tensor> outputs = prepareOutputs(tc_, "matmul", inputs);
+  auto timings = profile(*pExecutor, inputs, outputs);
+
+  auto inputDLTensors = makeDLConstTensors(inputs);
+  auto outputDLTensors = makeDLTensors(outputs);
+  optionsCache.recordRuntime(
+      lang::canonicalTc(tc_),
+      makeTensorInfoVector(extractRawPtrs(inputDLTensors)),
+      makeTensorInfoVector(extractRawPtrs(outputDLTensors)),
+      CudaGPUInfo::GPUInfo().getCudaDeviceStr(),
+      options,
+      timings.kernelRuntime);
+
+  {
+    options = CudaMappingOptions::makeNaiveMappingOptions();
+    auto pExecutor = compile<CudaBackend>(tc_, "matmul", inputs, options);
+    auto timings = profile(*pExecutor, inputs, outputs);
+    optionsCache.recordRuntime(
+        lang::canonicalTc(tc_),
+        makeTensorInfoVector(extractRawPtrs(inputDLTensors)),
+        makeTensorInfoVector(extractRawPtrs(outputDLTensors)),
+        CudaGPUInfo::GPUInfo().getCudaDeviceStr(),
+        options,
+        timings.kernelRuntime);
+  }
 
   FLAGS_tuner_gen_restore_number = 2;
```

Comment:
flags are global variables, we should really stop using them this way; but in a separate PR to make this clear to everyone

Reply:
yup .. legacy